browser/ui/webui/brave_rewards_ui.cc

```diff
@@ -1295,6 +1306,90 @@ void RewardsDOMHandler::FetchBalance(const base::ListValue* args) {
+      statement.SetStringKey("percentage", std::to_string(item.percentage));
+      statement.SetBoolKey("verified", item.verified);
+      statement.SetBoolKey("excluded", item.excluded);
+      statement.SetStringKey("name", item.name);
+      statement.SetStringKey("provider", item.provider);
+      statement.SetStringKey("url", item.url);
+      statement.SetStringKey("faviconUrl", item.favicon_url);
+      statement.SetStringKey("date", std::to_string(item.date));
+      statement.SetIntKey("category", item.category);
+      statement.SetStringKey("probi", std::to_string(item.probi));
+      statement.SetStringKey("reconcileStamp",
+          std::to_string(item.reconcile_stamp));
+      statements.GetList().push_back(std::move(statement));
+    }
+    monthly.SetKey("statementItems", base::Value(std::move(statements)));
+
+    base::Value transactions(base::Value::Type::LIST);
+    for (const auto& tx : monthly_statement->transactions) {
+      base::Value transaction(base::Value::Type::DICTIONARY);
+      transaction.SetStringKey("probi", tx.probi);
+      transaction.SetStringKey("date", std::to_string(tx.date));
+      transaction.SetIntKey("category", tx.category);
+      transactions.GetList().push_back(std::move(transaction));
+    }
+    monthly.SetKey("transactions", base::Value(std::move(transactions)));
+
+    base::Value report(base::Value::Type::DICTIONARY);
+    report.SetStringKey("opening", monthly_statement->report.opening_balance);
+    report.SetStringKey("closing", monthly_statement->report.closing_balance);
+    report.SetStringKey("grant", monthly_statement->report.grants);
+    report.SetStringKey("ads", monthly_statement->report.earning_from_ads);
+    report.SetStringKey("contribute",
+        monthly_statement->report.auto_contribute);
+    report.SetStringKey("donation",
+        monthly_statement->report.recurring_donation);
+    report.SetStringKey("tips", monthly_statement->report.one_time_donation);
+    report.SetStringKey("total", monthly_statement->report.total);
+    monthly.SetKey("balanceReport", base::Value(std::move(report)));
+
+    base::Value months(base::Value::Type::LIST);
+    for (const std::string& month : monthly_statement->months_available) {
+      months.GetList().push_back(base::Value(month));
+    }
+    monthly.SetKey("monthsAvailable", base::Value(std::move(months)));
+
+    monthly.SetStringKey("reconcileStamp",
+        std::to_string(monthly_statement->reconcile_stamp));
+
+    web_ui()->CallJavascriptFunctionUnsafe(
+         "brave_rewards.onGetMonthlyStatements",
+         std::move(monthly));
+  }
+}
+
+void RewardsDOMHandler::GetMonthlyStatements(const base::ListValue* args) {
+  CHECK_EQ(2U, args->GetSize());
+  if (!rewards_service_) {
+    return;
+  }
+  std::string month;
+  args->GetString(0, &month);
+  std::string year;
+  args->GetString(1, &year);
+  int32_t month_conv;
+  uint32_t year_conv;
+  base::StringToInt(month, &month_conv);
```

Comment:
Could we send these as integers from JavaScript?

Reply:
Done.